pkg/planner/core/indexmerge_path.go

```diff
@@ -94,7 +98,12 @@ func (ds *DataSource) generateIndexMergePath() error {
 	}
 
 	// If len(indexMergeHints) > 0 and some index-merge paths were added, then prune all other non-index-merge paths.
-	ds.possibleAccessPaths = ds.possibleAccessPaths[regularPathCount:]
+	// if len(ds.possibleAccessPaths) > oldIndexMergeCount, it means composed index merge path is generated, prune others.
```

Comment:
borrow from the old logic here, since composed cnf/dnf path is hard to be choosed as we said in the lark group
```
@Ryan Zhang 对，在没有 prune 之前，我想用这个 cost model 自己来判来着，结果发现 index plan 上累加 cost 的速度 >> table plan 上来减少的 row count 带来的代价减小
```